@xen-orchestra/fs/src/s3.js

```diff
@@ -50,35 +55,35 @@ export default class S3Handler extends RemoteHandlerAbstract {
       input.on('error', forwardError)
       inputStream = checksumStream
     }
-    const upload = this._s3.upload(
+    await this._s3.upload(
       {
         ...this._createParams(path),
         Body: inputStream,
       },
-      { partSize: IDEAL_FRAGMENT_SIZE }
+      { partSize: IDEAL_FRAGMENT_SIZE, queueSize: 1 }
     )
-    await upload.promise()
     if (checksum) {
       const checksum = await inputStream.checksum
       const params = {
         ...this._createParams(path + '.checksum'),
         Body: checksum,
       }
-      await this._s3.upload(params).promise()
+      await this._s3.upload(params)
     }
     await input.task
   }
 
   async _writeFile(file, data, options) {
-    return this._s3.putObject({ ...this._createParams(file), Body: data }).promise()
+    return this._s3.putObject({ ...this._createParams(file), Body: data })
   }
 
   async _createReadStream(file, options) {
-    return this._s3.getObject(this._createParams(file)).createReadStream()
+    // https://github.com/Sullux/aws-sdk/issues/11
+    return this._s3.getObject.raw(this._createParams(file)).createReadStream()
```

Comment:
Can you explain this change to me please?

Reply:
the `_s3` object is not the same before and after the change. after the change it becomes a wrapper of an AWS.S3 object whose error system has been improved and the promises unwrapped. 

But in that call we don't use a promise and the wrapper was falling us, calling `.raw()` gives us the previous behavior of receiving a request with all its methods.

Reply:
I see, my bad I did not follow the link, it's perfect as it is :slightly_smiling_face: